File: removing-stars-from-a-string/removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != '*') {
                stack.offer(s.charAt(i));
            }
            else {
                stack.pollLast();
            }
        }
        
        while (!stack.isEmpty()) {
            ans.append(stack.poll());
        }
        return ans.toString();
    }
}

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = deque()
        result = ''
        for char in s:
            if char != '*':
                stack.append(char)
            else:
                stack.pop()
        while stack:
            result += stack.popleft()
        return result